### PR TITLE
Detection of suspicious crontab creation or modification

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
https://github.com/elastic/detection-rules/issues/1934

## Summary
Detects the creation or modification of a crontab via a process that is not crontab (i.e python, osascript, etc.). This activity should not be highly prevalent and could indicate the use of cron as a persistence mechanism by a threat actor.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_crontab_creation.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+]
+risk_score = 47
+rule_id = "530178da-92ea-43ce-94c2-8877a826783d"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where event.type != "deletion" and file.path : "/private/var/at/tabs/*" and process.name != null and not process.name == "crontab"
```
Comment (DefSecSentinel) [inline rules/macos/persistence_crontab_creation.toml:29]:
```suggestion
file where event.type != "deletion" and process.name != null and file.path : "/private/var/at/tabs/*" and not process.executable == "/usr/bin/crontab"
```

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/macos/persistence_crontab_creation.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[metadata]
+creation_date = "2022/04/25"
+maturity = "development"
```
Comment (brokensound77) [inline rules/macos/persistence_crontab_creation.toml:3]: still in dev? If not then

```suggestion
maturity = "production"
```

File: rules/macos/persistence_crontab_creation.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+updated_date = "2022/04/25"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies attempts to create or modify a crontab via a process that is not crontab (i.e python, osascript, etc.). This
+activity should not be highly prevalent and could indicate the use of cron as a persistence mechanism by a threat actor.
+"""
+from = "now-9m"
+index = ["logs-*"]
```
Comment (brokensound77) [inline rules/macos/persistence_crontab_creation.toml:13]:
```suggestion
index = ["logs-endpoint.events.*"]
```

File: rules/macos/persistence_crontab_creation.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+]
+risk_score = 47
+rule_id = "530178da-92ea-43ce-94c2-8877a826783d"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where event.type != "deletion" and process.name != null and file.path : "/private/var/at/tabs/*" and not process.executable == "/usr/bin/crontab"
```
Comment (brokensound77) [inline rules/macos/persistence_crontab_creation.toml:29]: nit:

```suggestion
file where event.type != "deletion" and process.name != null and 
  file.path : "/private/var/at/tabs/*" and not process.executable == "/usr/bin/crontab"
```

**Review by Samirbous — APPROVED** [review]

**Review by brokensound77 — APPROVED** [review]

**Pull request merged**